toolchain/check/class.h

```diff
@@ -0,0 +1,32 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_CHECK_CLASS_H_
+#define CARBON_TOOLCHAIN_CHECK_CLASS_H_
+
+#include "toolchain/check/context.h"
+
+namespace Carbon::Check {
+
+// If `type_id` is a class type, get its corresponding `SemIR::Class` object.
+// Otherwise returns `nullptr`.
+auto TryGetAsClass(Context& context, SemIR::TypeId type_id) -> SemIR::Class*;
+
+// Builds the `Self` type using the resulting type constant.
+auto SetNewClassSelfTypeId(Context& context, SemIR::ClassId class_id) -> void;
+
+// Tracks that this declaration is the definition and introduce `Self`.
+auto TrackClassDefinition(Context& context, SemIR::ClassId class_id,
```

Comment:
I'm not sure what "tracks" means. If I understand correctly, this is intended to handle some logic tied to starting the class definition, and "introduce `Self`" is referring to the `AddRequiredName` call. Maybe:
```suggestion
// Starts the class definition, adding `Self` to name lookup.
auto StartClassDefinition(Context& context, SemIR::ClassId class_id,
```

Reply:
Done.